common/db/db.go

```diff
@@ -66,18 +69,57 @@ func CloseMongoConn() {
 func checkDBConnection() {
 	errorCounter := 0
 	for {
-		if errorCounter < 10 {
-			time.Sleep(1 * time.Second)
-			err := mongoClient.Ping(mongoClientCtx, nil)
-			if err != nil {
-				fmt.Println("FAILED PINGING MONGO")
-				errorCounter++
-				continue
+		time.Sleep(1 * time.Second)
+		err := mongoClient.Ping(mongoClientCtx, nil)
+
+		if err != nil {
+			errorCounter++
+			log.WithFields(log.Fields{
+				"error_count": errorCounter,
+				"error":       err,
+			}).Warn("Failed to ping MongoDB")
+
+			if err := reconnectMongo(); err != nil {
+				log.WithFields(log.Fields{
+					"error_count": errorCounter,
+					"error":       err,
+				}).Error("Failed to reconnect to MongoDB")
+
+				if errorCounter >= 10 {
+					log.Fatal("Lost connection to MongoDB server and failed to reconnect after 10 attempts!")
+				}
+			} else {
+				log.Info("MongoDB connection restored")
+				errorCounter = 0
 			}
 		} else {
-			log.Fatal("Lost connection to MongoDB server for more than 10 seconds!")
+			if errorCounter > 0 {
+				log.Info("MongoDB connection restored")
+				errorCounter = 0
+			}
+		}
```

Comment:
I am not sure this else statement is actually needed, you already set `errorCounter` to 0 on successful reconnect, when this block is reached it should already be in healthy state

Reply:
The else block was removed in the last commit